lib/node_modules/@stdlib/array/empty-like/docs/types/index.d.ts

```diff
@@ -324,473 +98,12 @@ declare function emptyLike( x: AnyArray, dtype: 'generic' ): Array<number>;
-* -   `float32`: single-precision floating-point numbers (IEEE 754)
-* -   `complex128`: double-precision complex floating-point numbers
-* -   `complex64`: single-precision complex floating-point numbers
-* -   `int32`: 32-bit two's complement signed integers
-* -   `uint32`: 32-bit unsigned integers
-* -   `int16`: 16-bit two's complement signed integers
-* -   `uint16`: 16-bit unsigned integers
-* -   `int8`: 8-bit two's complement signed integers
-* -   `uint8`: 8-bit unsigned integers
-* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
-* -   `generic`: generic JavaScript values
-*
-* @param x - input array from which to derive the output array length
-* @param dtype - data type
-* @returns empty array
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
-* var x = zeros( 2, 'generic' );
-* // returns [ 0.0, 0.0 ]
-*
-* var arr = emptyLike( x );
-* // returns [ 0.0, 0.0 ]
-*/
-declare function emptyLike( x: Array<any>, dtype?: DataType ): Array<number>;
-
-/**
-* Creates an uninitialized array having the same length and data type as a provided input array.
-*
-* ## Notes
-*
-* -   In browser environments, the function always returns zero-filled arrays.
-* -   If `dtype` is `'generic'`, the function always returns a zero-filled array.
-* -   In Node.js versions `>=3.0.0`, the underlying memory of returned typed arrays is **not** initialized. Memory contents are unknown and may contain **sensitive** data.
-*
-* The function recognizes the following data types:
-*
-* -   `float64`: double-precision floating-point numbers (IEEE 754)
-* -   `float32`: single-precision floating-point numbers (IEEE 754)
-* -   `complex128`: double-precision complex floating-point numbers
-* -   `complex64`: single-precision complex floating-point numbers
-* -   `int32`: 32-bit two's complement signed integers
-* -   `uint32`: 32-bit unsigned integers
-* -   `int16`: 16-bit two's complement signed integers
-* -   `uint16`: 16-bit unsigned integers
-* -   `int8`: 8-bit two's complement signed integers
-* -   `uint8`: 8-bit unsigned integers
-* -   `uint8c`: 8-bit unsigned integers clamped to `0-255`
-* -   `generic`: generic JavaScript values
-*
-* @param x - input array from which to derive the output array length
-* @param dtype - data type
-* @returns empty array
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
-* var x = zeros( 2, 'float32' );
-* // returns <Float32Array>[ 0.0, 0.0 ]
-*
-* var arr = emptyLike( x );
-* // returns <Float32Array>
-*
-* @example
-* var zeros = require( '@stdlib/array/zeros' );
-*
-* var x = zeros( 2, 'float64' );
-* // returns <Float32Array>[ 0.0, 0.0 ]
-*
-* var arr = emptyLike( x );
-* // returns <Float64Array>
-*/
-declare function emptyLike( x: AnyArray, dtype?: DataType ): AnyArray;
+declare function emptyLike<T extends keyof NumericAndGenericDataTypeMap<number>>( x: AnyArray, dtype: T ): NumericAndGenericDataTypeMap<number>[T];
```

Comment:
@Planeshifter Why not just use `DataTypeMap`? Why `NumericAndGeneric*`?

Reply:
This also isn't future proof, given the high likelihood of us add a boolean array variant.

Reply:
@kgryte They are equivalent right now, so switching is fine and shouldn't have any impact. I did it because the generic array here is typed to be only for numbers in this particular function signature and so it felt reasonable to make that constraint also more broadly in `NumericAndGeneric`; but yeah, deviates from `empty` now as I didn't have the additional type maps then...

Reply:
Makes sense. I used `DataType` intentionally when this was written, as an empty array need not be numeric. This is in contrast to `ones` and `zeros`. That `DataType` is currently equivalent to `NumericDataType | 'generic'` is an implementation detail we should not assume.